app/models/aspace/resource.rb

```diff
@@ -1,23 +1,40 @@
 # frozen_string_literal: true
 
 module Aspace
+  class AspaceResourceError < StandardError; end
+
   # Model for an Aspace resource to manage information
   # needed for downloading and updating resources in ArcLight
   class Resource
-    attr_reader :ead_id, :uri, :repository_code
+    attr_reader :ead_id, :uri, :identifier, :repository_code
 
-    def initialize(ead_id:, uri:, repository_code:)
+    def initialize(uri:, repository_code:, ead_id: nil, identifier: nil)
       @ead_id = ead_id
+      @identifier = identifier
       @uri = uri
       @repository_code = repository_code
     end
 
     def file_name
-      ead_id.sub(/(\.xml)?$/i, '').parameterize
+      resource_id.sub(/(\.xml)?$/i, '').parameterize
+    end
+
+    def resource_id
+      if ead_id
+        ead_id
+      elsif identifier
+        # Identifier can be entered in ASpace by archivists
+        # It often matches the EAD ID, but with variable punctionation or spacing
+        identifier
+      else
+        # If no EAD ID or identifier is provided, throw an error.
+        # We need an identifier to generate a file name and determine the repository code
+        raise AspaceResourceError, 'Resources from ArchivesSpace must have an EAD ID or identifier'
+      end
```

Comment:
Maybe? With your helpful comments above the method.
```suggestion
            ead_id || identifier || raise(AspaceResourceError,
                                    'Resources from ArchivesSpace must have an EAD ID or identifier')
```

Reply:
I think we will want to rescue from this error where it might be encountered in `DownloadEadJob` and log it to honeybadger so that if raised it doesn't interrupt queuing up the rest of the jobs.